### PR TITLE
CDATA parsing implemented, New Parsing Test Format

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Parser now considers CDATA while parsing, Node class modified to handle it as well. Also new Parsing test format that is easier to debug and cleaner.

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLHighlighting.java [hunk 1]
```diff
@@ -73,20 +111,27 @@ private static boolean covers(Range range, Position position) {
...
+
 		TokenType token = scanner.scan();
 		while (token != TokenType.EOS && token != tokenType) {
 			token = scanner.scan();
 		}
 		if (token != TokenType.EOS) {
 			try {
-				return new Range(xmlDocument.positionAt(scanner.getTokenOffset()),
-						xmlDocument.positionAt(scanner.getTokenEnd()));
+				return new Range(fmDocument.positionAt(scanner.getTokenOffset()),
```
Comment (angelozerr) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLHighlighting.java]: why fmDocument? It should be xmlDocument

Comment (NikolasKomonen) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLHighlighting.java]: you're right, I don't know why I even did that.

Comment (fbricon) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLHighlighting.java]: @NikolasKomonen you sjould make sure unit tests catch that

File: .gitignore [hunk 2]
```diff
@@ -25,6 +25,7 @@ target/
 *~
 
 .gradle
+.vscode/
```
Comment (fbricon) [inline .gitignore]: don't ignore .vscode

**Comment by NikolasKomonen** [comment]
@angelozerr would you be okay with me merging this, or do you have any comments?

**Comment by angelozerr** [comment]
Thanks @NikolasKomonen !

**Pull request merged**